src/vast_client.js

```diff
@@ -132,9 +175,25 @@ export class VASTClient {
         );
       }
 
-      this.vastParser
-        .getAndParseVAST(url, options)
-        .then((response) => resolve(response))
+      this.vastParser.initParsingStatus(options);
+      this.fetcher.setOptions(options);
+      this.vastParser.rootURL = url;
+
+      this.fetcher
+        .fetchVAST({
+          url,
+          emitter: this.vastParser.emit.bind(this.vastParser),
```

Comment:
Can we just extend event emitter on fetcher ? If it's not VASTParser responsability to propagate the events in fetchVAST, then it shouldn't need it. We would avoid these bind things.